From 9c80cab12c6fbd12396e4661838fb755cd1b3cde Mon Sep 17 00:00:00 2001
From: Robin Kauffman <robink@creosotehill.org>
Date: Wed, 5 Jun 2019 12:13:52 -0700
Subject: [PATCH 1/3] Apply srcshelton's ebtables-save patch.

Had to be reapplied by hand, as the patch no longer matched the source
closely enough to be applied by either patch or 'git patch'.
---
 ebtables-save.in | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/ebtables-save.in b/ebtables-save.in
index 17924a2..1dae57a 100644
--- a/ebtables-save.in
+++ b/ebtables-save.in
@@ -12,6 +12,7 @@ my $ebtables = "@sbindir@/ebtables";
 my $cnt = "";
 my $version = "1.0";
 my $table_name;
+my @table_names;
 
 # ========================================================
 # Process filter table
@@ -49,12 +50,19 @@ sub process_table {
 }
 # ========================================================
 
+if ($#ARGV + 1 == 0) {
+   @table_names =split("\n", `grep -E '^ebtable_' /proc/modules | cut -f1 -d' ' | sed s/ebtable_//`);
+}
+else {
+   @table_names = @ARGV;
+}
+# ========================================================
 unless (-x $ebtables) { exit -1 };
 print "# Generated by ebtables-save v$version (legacy) on " . `date`;
 if (defined($ENV{'EBTABLES_SAVE_COUNTER'}) && $ENV{'EBTABLES_SAVE_COUNTER'} eq "yes") {
     $cnt = "--Lc";
 }
-foreach $table_name (split("\n", `grep -E '^ebtable_' /proc/modules | cut -f1 -d' ' | sed s/ebtable_//`)) {
+foreach $table_name (@table_names) {
     $table =`$ebtables -t $table_name -L $cnt`;
     unless ($? == 0) { print $table; exit -1 };
     &process_table($table);
-- 
2.20.1

